lib/block/blockupload.js

```diff
@@ -12,10 +12,107 @@
+     * @property {Number} fileSize Size of the file to upload
+     * @property {String} filePath Path on the local disk to upload
+     * @property {Blob} [blob] Browser blob to upload (instead of fileUrl)
+     * @property {Boolean} [createVersion=false] Create version on duplicates
+     * @property {String} [versionLabel] Version label to apply to the created/updated file
+     * @property {String} [versionComment] Version comment to apply to the created/updated file
+     * @property {Boolean} [replace=false] True if the existing file should be replaced
+     */
+    /**
+     * @typedef {Object} BlockUploadOptions
+     * @property {UploadFile[]} uploadFiles List of files that will be uploaded to the target URL. 
+     * @property {*} headers HTTP headers that will be included in each request sent to AEM
+     * @property {Boolean} concurrent If true, multiple files in the supplied list of upload files will transfer simultaneously. If false, only one file will transfer at a time, and the next file will not begin transferring until the current file finishes.
+     * @property {Number} maxConcurrent Maximum number of concurrent HTTP requests that are allowed
+     * @property {Number} [preferredPartSize] Preferred part size
+     */
+    /**
+     * Upload files to AEM
+     * 
+     * @param {BlockUploadOptions} options AEM upload options
+     */
+    async uploadFiles(options) {
+        const preferredPartSize = options && options.preferredPartSize;
+        const maxConcurrent = (options && options.concurrent && options.maxConcurrent) || 1;
+
+        const controller = new TransferController();
+        controller.on(TransferEvents.AEM_INITIATE_UPLOAD, transferEvent => {
+            this.emit("filestart", transferEvent.transferAsset.eventData);
+        });
+        controller.on(TransferEvents.JOIN_TRANSFER_PARTS, transferEvent => {
+            this.emit("fileprogress", {
+                ...transferEvent.transferAsset.eventData,
+                transferred: transferEvent.props.transferBytes
+            });
+        });
+        controller.on(TransferEvents.AFTER_AEM_COMPLETE_UPLOAD, transferEvent => {
+            this.emit("fileend", transferEvent.transferAsset.eventData);
+        });
+        controller.on(TransferEvents.ERROR, transferEvent => {
+            if (transferEvent.props.firstError) {
+                this.emit("fileerror", {
+                    ...transferEvent.transferAsset.eventData,
+                    errors: [ UploadError.fromError(transferEvent.error) ]
+                });
+            }
+        });
+
+        const retryOptions = {
+            retryMaxCount: 5
+        };
+        
+        // Build and execute pipeline
+        const randomFileAccess = new RandomFileAccess();
+        try {
+            const pipeline = new Pipeline(
+                new FailUnsupportedAssets(),
+                new MapConcurrent(new AEMInitiateUpload(retryOptions), { maxBatchLength: 100 }),
```

Comment:
I thought the idea was to go without the AEM implementation?

Reply:
true, this will change in my next commit. this commit to just share the test files, in case you needed them.